### PR TITLE
Bump to forc 0.44.1

### PR DESCRIPTION
**Pull request by bitzoic**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)

## Changes

The following changes have been made:

- Bumped libraries to forc v0.44.1
- Bumped libraries to fuel-core v0.20.3
- Bumped libraries to rust v0.71.1
- Bump libraries to fuels-rs v0.46.0
- Archived the NFT library due to the introduction of the token standard 

## Notes

- A new NFT library will need to be created based on the new standard and multi-asset functionality.
- The fuel-core updates to the fuel-merkle repo will be done in a separate PR here: https://github.com/FuelLabs/sway-libs/issues/186

### CODE REVIEW

File: libs/archive/nft/README.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
     </picture>
 </p>
 
+> **Warning**
+> This library has been deprecated with the introduction of [Sway multi-token standard](https://github.com/FuelLabs/sway-standards/issues/1). Use of this library is *highly* discouraged. The last supported version is forc v0.40.1.g
```
Comment (Braqzen) [inline libs/archive/nft/README.md:9]:
```suggestion
> This library has been deprecated with the introduction of [Sway multi-token standard](https://github.com/FuelLabs/sway-standards/issues/1). Use of this library is *highly* discouraged. The last supported version is forc v0.40.1.

```

File: tests/src/fixed_point/ufp32_test/src/main.sw
Comment (Braqzen) [inline tests/src/fixed_point/ufp32_test/src/main.sw:1]: Haven't confirmed the conversions here

**Review by Braqzen — COMMENTED** [review]
I wouldn't archive by removing it from CI. I'd keep it there and keep bumping but note that the standard is out of date and should not be followed.

This at least shows the up to date code

**Comment by bitzoic** [comment]
> I wouldn't archive by removing it from CI. I'd keep it there and keep bumping but note that the standard is out of date and should not be followed.
> 
> This at least shows the up to date code

This is very far off from what we would like to promote as good Sway code. I've opted to leave it only so that existing projects are able to reference it. As soon as https://github.com/FuelLabs/sway-libs/issues/187 is closed it will be removed entirely.

**Comment by Braqzen** [comment]
> > I wouldn't archive by removing it from CI. I'd keep it there and keep bumping but note that the standard is out of date and should not be followed.
> > This at least shows the up to date code
> 
> This is very far off from what we would like to promote as good Sway code. I've opted to leave it only so that existing projects are able to reference it. As soon as #187 is closed it will be removed entirely.

If we are not maintaining it and it's bad code why not just remove it now?

**Review by K1-R1 — APPROVED** [review]

**Pull request merged**